pages/docs/tutorials/native/dynamic-libraries.md

```diff
@@ -0,0 +1,339 @@
+`demo_KNativePtr` is defined as `void*` above. 
+
+There is no namespaces support in C, so that Kotlin/Native compiler generates 
+long names to avoid a possible clash with other symbols of your native project.
+
+The most significant part of definitions goes further in the `demo_api.h` file.
+It includes the definition of our Kotlin/Native library world:
+
+```c
+typedef struct {
+  /* Service functions. */
+  void (*DisposeStablePointer)(demo_KNativePtr ptr);
+  void (*DisposeString)(const char* string);
+  demo_KBoolean (*IsInstance)(demo_KNativePtr ref, const demo_KType* type);
+
+  /* User functions. */
+  struct {
+    struct {
+      struct {
+        void (*ints)(demo_KByte b, demo_KShort s, demo_KInt i, demo_KLong l);
+        void (*floats)(demo_KFloat f, demo_KDouble d);
+        const char* (*strings)(const char* str);
+        const char* (*get_globalString)();
+        struct {
+          demo_KType* (*_type)(void);
+          demo_kref_demo_DemoClazz (*DemoClazz)();
+          demo_KLong (*foo)(demo_kref_demo_DemoClazz thiz, demo_KInt p);
+        } DemoClazz;
+      } demo;
+    } root;
+  } kotlin;
+} demo_ExportedSymbols;
+```
+
+The code uses anonymous structure declarations. The code `struct { .. } foo`
+declares a field in the outer struct of that 
+anonymous structure type, the type with no name. 
+
+C does not support objects too. People use function pointers to mimic 
+object semantics. A function pointer is declared as follows `RETURN_TYPE (* FIELD_NAME)(PARAMETERS)`.
+It reads tricky, but we should be able to see function pointer fields in the structures above. 
+
+### Runtime Functions
+
+The code reads as follows. We have the `demo_ExportedSymbols` structure which defines
+all functions that Kotlin/Native and our library provides to us. It heavily uses 
+nested anonymous structures to mimic packages.
+
+The `demo_ExportedSymbols` structure contains several helper functions:
+```c
+void (*DisposeStablePointer)(demo_KNativePtr ptr);
+void (*DisposeString)(const char* string);
+demo_KBoolean (*IsInstance)(demo_KNativePtr ref, const demo_KType* type);
+```
+
+Those functions deals with Kotlin/Native objects. One calls 
+`DisposeStablePointer` to release a Kotlin object and `DisposeString` to release Kotlin String, 
+which has the `char*` type in C. One may use the `IsInstance` function to check, if a
+Kotlin type, a `demo_KNativePtr` is an instance of another type. 
+ 
+Kotlin/Native has garbage collection, but it does not help to deal
+with Kotlin objects from C language. Kotlin/Native has an interop with Objective-C and 
+Swift, and integrates with their reference counters. You may want to find more details
+in the [documentation](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md)
+or to check the related [TUTORIAL_KOTLIN_macOS_FRAMEWORK_LINK].
+
+### Our Library Functions
+
+Let's take a look on the nests structure with path `kotlin.root.demo`. The path 
```

Comment:
What is "on the nests structure"?

Reply:
I'll use `field` instead to refer to a chain of fields. It should be easier to understand, I guess